### DISCUSSION TITLE
sbt plugin ⇒ Sonatype OSS / Maven Central migration tracker

### DISCUSSION THREAD

**Discussion by eed3si9n** [sbt/sbt#7209] [General]
@mkurz wrote in https://github.com/sbt/sbt/issues/7202
>  I took a look at the `~/.cache/coursier/v1/https/repo.scala-sbt.org` folder of my machine to find out which libraries on my machine are currently hosted only on repo.scala-sbt.org and would therefore be affected by a total outage. This is what I came up with:

* https://github.com/sbt/sbt-dynver/issues/239
* https://github.com/sbt/sbt-license-report/issues/51
* https://github.com/sbt/sbt-sdlc/issues/3
* https://github.com/sbt/sbt-multi-jvm/issues/59
* https://github.com/sbt/sbt-osgi/issues/77
* https://github.com/cb372/sbt-explicit-dependencies/issues/102
* https://github.com/johanandren/sbt-akka-version-check/issues/1
* https://github.com/sbt/sbt-javaagent/issues/24
* https://github.com/spray/sbt-revolver/issues/112
* https://github.com/lightbend/sbt-publish-rsync/issues/6
* https://github.com/sbt/sbt-boilerplate/issues/32
* jamesward/play-auto-refresh#47

**Comment by eed3si9n** [comment]
If we are going to do this, I'd recommend using sbt 1.9.0-M1 so the Maven Central (re)publishing would be POM consistent. /cc @adpi2

**Comment by eed3si9n** [comment]
sbt-license-report 1.3.0 - https://github.com/sbt/sbt-license-report/releases/tag/v1.3.0 is out.

**Comment by eed3si9n** [comment]
sbt-sdlc 0.3.0 https://github.com/sbt/sbt-sdlc/releases/tag/v0.3.0 is out.

**Comment by eed3si9n** [comment]
sbt-dynver 5.0.0 https://github.com/sbt/sbt-dynver/releases/tag/v5.0.0 is out.

**Comment by eed3si9n** [comment]
sbt-pom-reader 2.4.0 https://github.com/sbt/sbt-pom-reader/releases/tag/v2.4.0 is out.

**Comment by mkurz** [comment]
@eed3si9n Please add
* https://github.com/jamesward/play-auto-refresh/issues/47
to the above list, thanks!

**Comment by julienrf** [comment]
We had access to the stats showing the most downloaded plugins from the community repository, and here is the list, sorted by descending number of downloads per month:

- [x] sbt-compat https://github.com/dwijnand/sbt-compat/issues/66 Dale back-published v1.2.6 to Sonatype.
- [x] sbt-web https://github.com/sbt/sbt-web/pull/194
- [x] sbt-js-engine https://github.com/sbt/sbt-js-engine/pull/89
- [x] sbt-twirl https://github.com/playframework/twirl/pull/435 (since 1.5.2)
- [x] sbt-javaagent https://github.com/sbt/sbt-javaagent/issues/23 https://github.com/sbt/sbt-javaagent/pull/26
- [x] sbt-revolver https://github.com/spray/sbt-revolver/pull/114
- [x] sbt-native-packager. Already migrated. Version 1.5.2 was used by Play so far. Will be fixed in Play 2.9.x
- [x] sbt-git. Has been migrated but we still see many incoming requests (~1 million/month) to version 1.0.0 because that version is used by unmaintained plugins.
- [x] sbt-buildinfo https://github.com/sbt/sbt-buildinfo/pull/195
- [x] sbt-assembly Migrated, but versions 0.14.x and 0.15.0 are still used a lot by old versions of sbt-buildo and sbt-s3-publisher, and by projectile and sbt-reproducible-builds

**Reply by mkurz** [reply]
sbt-buildinfo get published on maven central already:
* https://repo1.maven.org/maven2/com/eed3si9n/sbt-buildinfo_2.12_1.0/0.11.0/
* https://github.com/sbt/sbt-buildinfo/pull/195

**Reply by mkurz** [reply]
sbt-assembly also, but it misses the GitHub action workflow using sbt-ci-release:
* https://repo1.maven.org/maven2/com/eed3si9n/sbt-assembly_2.12_1.0/2.1.1/ :+1: 
* https://github.com/sbt/sbt-assembly/tree/develop/.github/workflows :-1:

**Reply by mkurz** [reply]
sbt-revolver migrated already as well:
* https://github.com/spray/sbt-revolver/pull/114
* https://github.com/spray/sbt-revolver/commit/4d28f091d3f2ce2f7f28b0165bac6705effded9a
* https://repo1.maven.org/maven2/io/spray/sbt-revolver_2.12_1.0/0.10.0/

**Reply by julienrf** [reply]
Interesting, thank you very much for all the pointers!

**Reply by mzuehlke** [reply]
Thanks @julienrf 
So the top 10 requested plugins have already been published to Maven Central, (you are on the last one missing)
But many projects are still using an older version 😢
